lib/cocina/models/builders/title_builder.rb

```diff
@@ -24,24 +24,46 @@ def self.build(titles, strategy: :first, add_punctuation: true)
           new(strategy: strategy, add_punctuation: add_punctuation).build(titles)
         end
 
+        # the "main title" is the title withOUT subtitle, part name, etc.  We want to index it separately so
+        #   we can boost matches on it in search results (boost matching this string higher than matching full title string)
+        #   e.g. "The Hobbit" (main_title) vs "The Hobbit, or, There and Back Again (full_title)
+        # @param [[Array<Cocina::Models::Title,Cocina::Models::DescriptiveValue>] titles the titles to consider
+        # @param [Symbol] strategy ":first" is the strategy for selection when primary or display
+        #   title are missing
+        # @param [Boolean] add_punctuation determines if the title should be formmated with punctuation
+        # @return [String] the title value for Solr
+        def self.main_title(titles)
+          # 245a = title.structuredValue.value with type 'main title' (or just title.value),
+          # 245b = title.structuredValue.value with type 'subtitle'; if there are multiple titles, the 245 should have status 'primary' or at least be first
+          new(strategy: :first, add_punctuation: false).main_title(titles)
```

Comment:
Is the `:first` strategy used when invoking `#main_title`, or is it primarily used in `#build`?

Reply:
In other words, I wonder if `:main` or `:main_title` could or should be a strategy all its own and be a branch within `#build` instead of its own new instance method.

Reply:
build will be changing in my next PR;  I'm not sure I want to optimize the code until I've got the correct strings for main_title, build, full_title (next PR) and additional_titles (next next PR)

Reply:
Note that `build` is giving bad titles now in how it marries the pieces together for structured values.

Reply:
Gotcha, thanks!

Reply:
to answer your specific question, "first" will be used by main_title and full_title but "all" will be used by additional title